### PR TITLE
[FLINK-18072][hbase] HBaseLookupFunction can not work with new internal data structure RowData

### PR DESCRIPTION
**Pull request by leonardBang**
## What is the purpose of the change

* This PR fix HBase lookup function can not work properly with the new internal Data structure `RowData`.

## Brief change log

  - Import org/apache/flink/connector/hbase/source/HBaseRowDataLookupFunction.java to lookup RowData
  - Clean up some test code

## Verifying this change

This change is covered by ITCase `HBaseConnectorITCase`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/HBaseConnectorITCase.java [hunk 1]
```diff
@@ -558,71 +539,71 @@ public void testHBaseLookupTableSource() throws Exception {
...
+			.map(Row::toString)
+			.sorted()
+			.collect(Collectors.toList());
+
+		// check result, the time type in collected result is LOCAL_DATE_TIME, LOCAL_DATE, LOCAL_TIME
+		List<String> expected = new ArrayList<>();
+		expected.add("1,1,10,Hello-1,100,1.01,false,Welt-1,2019-08-18T19:00,2019-08-18,19:00");
+		expected.add("2,2,20,Hello-2,200,2.02,true,Welt-2,2019-08-18T19:01,2019-08-18,19:01");
+		expected.add("3,2,30,Hello-3,300,3.03,false,Welt-3,2019-08-18T19:02,2019-08-18,19:02");
+		expected.add("3,3,30,Hello-3,300,3.03,false,Welt-3,2019-08-18T19:02,2019-08-18,19:02");
```
Comment (wuchong) [inline flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/HBaseConnectorITCase.java:586]: Could you add a DECIMAL to the lookup table? It is the other error-prone type.

File: flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/HBaseConnectorITCase.java [hunk 2]
```diff
@@ -377,57 +344,28 @@ public void testTableSink() throws Exception {
 			TableSink tableSink = TableFactoryService
 				.find(HBaseTableFactory.class, descriptorProperties.asMap())
 				.createTableSink(descriptorProperties.asMap());
-			((TableEnvironmentInternal) tEnv).registerTableSinkInternal("hbase", tableSink);
+			((TableEnvironmentInternal) tEnv).registerTableSinkInternal(TEST_TABLE_2, tableSink);
```
Comment (wuchong) [inline flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/HBaseConnectorITCase.java:347]: Can we also migrate this to use DDL?

File: flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/HBaseConnectorITCase.java [hunk 3]
```diff
@@ -455,99 +392,82 @@ public void testTableSink() throws Exception {
 
 	@Test
 	public void testTableSourceSinkWithDDL() throws Exception {
+		// only test TIMESTAMP/DATE/TIME/DECIMAL for new connector(using blink-planner), because new connector encodes
+		// DATE/TIME to int, the old one encodes to long, and DECIMAL with precision works well in new connector.
+		final boolean testTimeAndDecimalTypes = BLINK_PLANNER.equals(planner) && !isLegacyConnector;
```
Comment (wuchong) [inline flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/HBaseConnectorITCase.java:397]: This test code is really hard to maintain and read. I would suggest only test for blink planner and new connector for simplification.

**Comment by wuchong** [comment]
The failed case is not related to this PR.

**Comment by rmetzger** [comment]
Note, this change broke the release-1.11. I reverted the commit

**Comment by rmetzger** [comment]
But only in `release-1.11`: https://github.com/apache/flink/commit/8192e1bcc2f4fb5917c041a454bf9e20eda5b83e

**Comment by rmetzger** [comment]
This is the CI build: https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=3668&view=results

**Comment by leonardBang** [comment]
Thanks  @rmetzger , sorry for the broke,  ,the PR for release-1.11 has opened: https://github.com/apache/flink/pull/12691

**Pull request merged**